### PR TITLE
Add react redux

### PR DESCRIPTION
**Pull request by MahdiRezaeiDev**
### Book Store project: add react-redux package
in this branch, I have made the following changes:

1. Added the `react-redux` package.
2. Added the `<Provider>` content to the project.
3. Added the `useSelector` hook to get a books list from the global store.
4. Added the `useDispatch` hook to dispatch add new book action.
5. Added the `useDispatch` hook to dispatch remove book action.
6. Added the `useDispatch` hook to dispatch check category action.

### CODE REVIEW

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @MahdiSohaily  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that ** has no professional readme**. Please add a professional readme according to the template provided by microverse.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,43 +1,37 @@
...
-    let newState = [...state]
-    newState = newState.splice(action.index, 1)
-    return [...newState]
-  }
-}
+    let newState = [...state];
+    newState = newState.splice(action.index, 1);
+    return [...newState];
+  },
+};
```
Comment (zieeco) [inline src/redux/books/books.js]: - Hey, please, kindly improve on the following:

   - Your default book gets deleted correctly but if I delete them and add like 4 or 2 books, on deleting one, all the other books get deleted. Please test to see it in action so you can improve on this. :pray:

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @MahdiSohaily,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Good commit messages. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @MahdiSohaily,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**